src/fs.rs

```diff
@@ -0,0 +1,26 @@
+use camino::{Utf8Path, Utf8PathBuf};
+use std::fs;
+use std::io;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
```

Comment:
This seems unnecessary, since we have `impl From<anyhow::Error> for SetupError`. Our functions below could simply use `anyhow::Error` instead, since we will never dispatch on the variants of this enum anyway.

Reply:
Yes, true. l like this style, since it does not require us to add the necessary context about the `path` at the call site of the function. I makes the control flow a bit more visible in my opinion, and also reduces duplication.

Reply:
What I meant is this:
```
pub fn create_dir_all<P: AsRef<Utf8Path>>(path: P) -> anyhow::Result<()> {
    fs::create_dir_all(path.as_ref()).context(...)
}
```
Wouldn't this be sufficient?

Reply:
What I meant is this:
```
pub fn create_dir_all<P: AsRef<Utf8Path>>(path: P) -> anyhow::Result<()> {
    fs::create_dir_all(path.as_ref()).context(...)
}
```
Wouldn't this be sufficient?